qabelbox/src/test/java/de/qabel/qabelbox/storage/notifications/StorageNotificationTest.java

```diff
@@ -0,0 +1,110 @@
+        shadowOf(notificationManager).cancelAll();
+        storageNotificationManager = new StorageNotificationManager(
+                RuntimeEnvironment.application.getApplicationContext());
+    }
+
+    @Test
+    public void testSingleUpload() {
+        BoxUploadingFile uploadingFile = new BoxUploadingFile("FILE", UPLOAD_PATH, TEST_OWNER);
+        Queue<BoxUploadingFile> queue = new LinkedList<>();
+        queue.add(uploadingFile);
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+
+        queue.poll();
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+    }
+
+    @Test
+    public void testMultipleUpload() {
+        Queue<BoxUploadingFile> queue = new LinkedList<>();
+        queue.add(new BoxUploadingFile("FILE", UPLOAD_PATH, TEST_OWNER));
+        queue.add(new BoxUploadingFile("FILE2", UPLOAD_PATH, TEST_OWNER));
+        queue.add(new BoxUploadingFile("FILE3", UPLOAD_PATH, TEST_OWNER));
+        //3 Uploads
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+
+        //2 Uploads
+        queue.poll();
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+        //1 Upload
+
+        queue.poll();
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+
+        //Uploads complete
+        queue.poll();
+        storageNotificationManager.updateUploadNotification(queue);
+        checkNotification(queue);
+    }
+
+    private void checkNotification(Queue<BoxUploadingFile> queue) {
+        BoxUploadingFile currentUpload = queue.peek();
+
+        ShadowNotificationManager shadowManger = shadowOf(notificationManager);
+        assertEquals(1, shadowManger.size());
+        Notification notification = shadowManger.getAllNotifications().get(0);
+        ShadowNotification shadowNotification = shadowOf(notification);
+
+        if (currentUpload != null) {
+            int expectedProgress = currentUpload.getUploadStatusPercent();
+            assertEquals(expectedProgress, shadowNotification.getProgressBar().getProgress());
+        }
+
+        String expectedTitle = (queue.size() > 0 ?
+                RuntimeEnvironment.application.getResources().
+                        getQuantityString(R.plurals.uploadsNotificationTitle, queue.size(), queue.size())
+                : RuntimeEnvironment.application.getString(R.string.upload_complete_notification_title));
+        assertEquals(expectedTitle, shadowNotification.getContentTitle());
+
+        String expectedContent = "";
+        if (currentUpload != null) {
+            expectedContent = String.format(RuntimeEnvironment.application.getString(R.string.upload_in_progress_notification_content), currentUpload.name);
+        }
+        assertEquals(expectedContent, shadowNotification.getContentText());
```

Comment:
All assertions on the content of the notification should be done in a separate test.